core/src/main/java/org/apache/iceberg/view/ViewVersionReplace.java

```diff
@@ -56,6 +57,14 @@ public ViewVersion apply() {
   }
 
   ViewMetadata internalApply() {
+    // Replacing a materialized view is not supported because the old storage location will wrongly
+    // transfer to the new version
+    // if not handled properly.
```

Comment:
I think we will want to allow this by adding the view version ID to metadata in the table. If you load the view, then load the table and the version doesn't match between them then the table cannot be used.

Reply:
Agreed, just keeping it out of scope of this PR. Let me know if we should keep in this PR's scope. (also agree that we can change the comment in case we leave it out of scope).

Reply:
Since this relates to the spec, I have decided to add this support. Now there is a new view version property `materialized.view.version.id` that is tracked at the table level and it is factored into freshness evaluation.